docs/metamacros.md

```diff
@@ -0,0 +1,66 @@
+### Docile Metamacros
+
+Docile uses an extensible metamacro syntax within docstrings to support greater flexibility.
+
+#### The syntax constist of four required parts:
+
+* `!!` double qestionmark
+* followed by `more than 0 chars` which must be a valid julia identifier
+* `(` opening bracket - parentheses
+* `)` closing bracket - parentheses
+
+Optional there may be text between opening and closing parentheses.
+
+
+Example:
+
+```julia
+"""
+
+!!hypothetical()
+
+!!set(author:Hugo Smith)
+
+!!get(author)
+
+!!setget(笔者:所以不多说了)
+
+!!summary(Set the value for a field in an object's metadata.)
+
+!!longform(
+...
+)
+
+"""
+```
```

Comment:
I skipped that because there might be still some changes and later we might prefer to just link to the section.